### PR TITLE
added filesystem snapshot resource

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
A few sentences describing the overall goals of the pull request's commits.
Added file system snapshot 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

image

image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests
- [x]  Unit tests

### CODE REVIEW

File: powerstore/resource_filesystem_snapshot.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	}
+
+	fileSystemID := plan.FileSystemID.ValueString()
+	name := plan.Name.ValueString()
+	description := plan.Description.ValueString()
+	expirationTimestamp := plan.ExpirationTimestamp.ValueString()
+	accessType := plan.AccessType.ValueString()
+
+	// Create new filesystem snapshot
+	snapCreate := &gopowerstore.SnapshotFSCreate{
```
Comment (shenda1) [inline powerstore/resource_filesystem_snapshot.go:152]: Can we directly populate the struct from the plan?

File: examples/resources/powerstore_filesystem_snapshot/import.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+
+
+# Below are the steps to import snapshot :
+# Step 1 - To import a filesystem snapshot , we need the id of that filesystem snapshot
+# Step 2 - To check the id of the filesystem snapshot we can make Get request to filesystem snapshot endpoint. eg. https://10.0.0.1/api/rest/filesystem and 'filesystem_type: Snapshot' which will return list of all filesystem snapshots ids.
```
Comment (shenda1) [inline examples/resources/powerstore_filesystem_snapshot/import.sh]: file_system

File: examples/resources/powerstore_filesystem_snapshot/provider.tf [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+terraform {
+  required_providers {
+    powerstore = {
+      version = "1.1.0"
```
Comment (shenda1) [inline examples/resources/powerstore_filesystem_snapshot/provider.tf]: 1.2.0

File: examples/resources/powerstore_filesystem_snapshot/resource.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Commands to run this tf file : terraform init && terraform plan && terraform apply
+# Create, Update, Delete is supported for this resource
+# To import , check powerstore_filesystem_snapshot/import.tf for more info
+# filesystem_id is the required attribute to create volume snapshot.
```
Comment (shenda1) [inline examples/resources/powerstore_filesystem_snapshot/resource.tf]: file system

File: go.mod [hunk 5]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/hashicorp/terraform-plugin-framework v1.13.0
 	github.com/hashicorp/terraform-plugin-framework-validators v0.15.0
 	github.com/hashicorp/terraform-plugin-go v0.25.0
+	github.com/hashicorp/terraform-plugin-log v0.9.0
```
Comment (shenda1) [inline go.mod:13]: Take latest commit for gopowerstore

**Review by rounak-adhikary — DISMISSED** [review]

**Comment by Krishnan-Priyanshu** [comment]
Need to revert the change done on FSModify Type on gopowerstore: https://github.com/dell/gopowerstore/pull/162

File: go.mod (hunk 5 above)
Comment (Krishnan-Priyanshu) [reply go.mod:13]: Need to revert the changes done on FSModify Type, will take the latest commit after merge.

**Review by shenda1 — DISMISSED** [review]

File: go.mod (hunk 5 above)
Comment (Krishnan-Priyanshu) [reply go.mod:13]: Updated the gopowerstore version to the latest.

**Review by shenda1 — APPROVED** [review]

**Review by rounak-adhikary — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**